controllers/paddlejob_controller.go

```diff
@@ -195,6 +195,30 @@ func (r *PaddleJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	}
 
+	if len(pdj.Status.PS.Refs) == pdj.Spec.PS.Replicas && len(pdj.Status.Worker.Refs) == pdj.Spec.Worker.Replicas {
+		if pdj.Spec.Intranet == pdv1.Service {
+			if len(pdj.Status.PS.Refs)+len(pdj.Status.Worker.Refs) != len(svcs.Items) {
+				return ctrl.Result{}, nil
+			}
+		}
+		if err := r.Get(ctx, types.NamespacedName{Name: pdj.Name, Namespace: pdj.Namespace}, &corev1.ConfigMap{}); err == nil || !apierrors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+		cm := constructConfigMap(&pdj, childPods)
+		if cm == nil {
+			return ctrl.Result{Requeue: true}, nil
+		}
+		if err := ctrl.SetControllerReference(&pdj, cm, r.Scheme); err != nil {
+			log.Error(err, "make reference failed")
+			return ctrl.Result{Requeue: true}, nil
+		}
+		err := r.createResource(ctx, &pdj, cm)
+		if apierrors.IsConflict(err) {
+			return ctrl.Result{Requeue: true}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
 	if len(pdj.Status.PS.Refs) == pdj.Spec.PS.Replicas && len(pdj.Status.Worker.Refs) == pdj.Spec.Worker.Replicas {
 		if err := r.Get(ctx, types.NamespacedName{Name: pdj.Name, Namespace: pdj.Namespace}, &corev1.ConfigMap{}); err == nil || !apierrors.IsNotFound(err) {
 			return ctrl.Result{}, err
```

Comment:
Seems merging is not working here?  Merge will introduce 2 if blocks, while only 1 is intended? @kuizhiqing